cpp/ycm/tests/CMakeLists.txt

```diff
@@ -33,6 +33,11 @@ if ( USE_SYSTEM_GMOCK )
   find_package( GTest REQUIRED )
   find_package( GMock REQUIRED )
 else()
+  if ( WIN32 )
+    # Override BUILD_SHARED_LIBS option in gmock and gtest CMakeLists
+    set( BUILD_SHARED_LIBS ON CACHE BOOL "Build shared libraries (DLLs)." )
```

Comment:
No need to spend too much time on it, but taking a look at the Flightcrew `CMakeLists.txt` files should make things much easier. It's all CMake, C++ and gtest/gmock. 

These three files should be enough to get an idea: [top-level](https://github.com/Sigil-Ebook/flightcrew/blob/master/CMakeLists.txt), [library](https://github.com/Sigil-Ebook/flightcrew/blob/master/src/FlightCrew/CMakeLists.txt), [tests](https://github.com/Sigil-Ebook/flightcrew/blob/master/src/FlightCrew/tests/CMakeLists.txt).